### ISSUE TITLE
error: 'Realm/RLMRealm_Private.h' file not found

### ISSUE THREAD

**Issue by mark-urbanthings** [realm/realm-swift#8164]
### How frequently does the bug occur?

Always

### Description

Having updated to 10.37 can no longer build our app, see log below. 

Issue does not occur with 10.36 or earlier.

### Stacktrace & log output

```shell
<module-includes>:45:9: note: in file included from <module-includes>:45:
#import "RLMAsyncTask_Private.h"
        ^
../Pods/Headers/Private/Realm/RLMAsyncTask_Private.h:21:9: error: 'Realm/RLMRealm_Private.h' file not found
#import <Realm/RLMRealm_Private.h>
        ^
../Pods/RealmSwift/RealmSwift/Aliases.swift:20:8: error: could not build Objective-C module 'Realm'
import Realm.Swift
       ^
```

### Can you reproduce the bug?

Always

### Reproduction Steps

This may be specific to our app, so at this time can't provide you with reproduction steps. 

### Version

10.37

### What Atlas Services are you using?

Local Database only

### Are you using encryption?

Yes

### Platform OS and version(s)

Build issue

### Build environment

Xcode version: Xcode 14.2 (14C18).
Dependency manager and version: CocoaPods 1.11.3

**Comment by dianaafanador3** [comment]
Hi @mark-urbanthings can you please do a clean cocoapods install, use `pod deintegrate`, remove your Podfile.lock and try again to `pod install` again. Clean your project derived data folder as well and try again. 
If you are still having issues, can you please attach your Podfile and check also check if this files are within the private header phase for the Realm target in the pods project, if not clearly something is going wrong with the cocoapods installation.

**Comment by mark-urbanthings** [comment]
Hi @dianaafanador3, 

I've tried the above steps with no success. Our project has complex dependencies where we are publishing our own libraries to private pod repo and integrating these alongside public pods. 

I've however managed to reproduce the issue with a much cut down test project. Issue seems to be tied to having an abstract project into which we include both Realm and our own library of Realm functionality (UTRealm). The development pod in attached zip of the project is again cut down representation of our UTRealm pod but is enough to reproduce the issue.

Regards
Mark
[BuildIssue.zip](https://github.com/realm/realm-swift/files/10948427/BuildIssue.zip)

**Comment by MunokKim** [comment]
Same problem to me at version 10.37

**Comment by mark-urbanthings** [comment]
@MunokKim are you using abstract_target as well?

**Comment by MunokKim** [comment]
@mark-urbanthings Sorry I don't understand about abstract_target

**Comment by mark-urbanthings** [comment]
@MunokKim in your Podfile (see [here](https://guides.cocoapods.org/syntax/podfile.html#abstract_target)). If you can post your Podfile here it may help diagnosing issue.

**Comment by dianaafanador3** [comment]
Added a PR https://github.com/realm/realm-swift/pull/8168 which may help solve this issue.

**Comment by mark-urbanthings** [comment]
@dianaafanador3 integrating from your PR branch fixes issue for our app as well as the test project.

**Comment by rpander93** [comment]
Running into the same build error and #8168 indeed resolves it for me!

**Comment by wangxue-holla** [comment]
try   needDynamicFramework = Array['RealmSwift', 'Realm']

  pre_install do |installer|
    installer.pod_targets.each do |pod|
      if needDynamicFramework.include?(pod.name)
        def pod.build_type;
        Pod::BuildType.dynamic_framework
      end
    end
  end
end

**Comment by maopenglin** [comment]
I have also encountered this problem, how to solve it

**Comment by dianaafanador3** [comment]
Fixed on version [v10.41.0](https://github.com/realm/realm-swift/releases/tag/v10.41.0)
